agent/agent.go

```diff
@@ -65,6 +65,13 @@ func buildCommandArgs(agentConfig config.AgentConfig) []string {
 	} else if agentConfig.EnableRelayModeForXds {
 		args = append(args, "--concurrency")
 		args = append(args, config.ENVOY_CONCURRENCY_FOR_RELAY_DEFAULT)
+
+		// If --log-level is not debug and not trace then set `aws` --component-log-level to debug.
+		// https://www.envoyproxy.io/docs/envoy/latest/operations/cli.html#cmdoption-component-log-level
+		if agentConfig.EnvoyLogLevel != "debug" && agentConfig.EnvoyLogLevel != "trace" {
+			args = append(args, "--component-log-level")
+			args = append(args, "aws:debug")
```

Comment:
From an initial glance it seems like this would cause additional logs beyond just credential fetching, like logs from signer_impl.cc. Your description makes it seem like only the credential fetching will show up. What keeps the additional logs from being included?

Reply:
oh yes, you are correct. it will also include logs related to signer with authorization headers which we don't want in the first place as well as it will log for every xDS request and response.

I guess I will close this PR now. In future if we want more insights into relay envoy logs then we can enhance on [enableLogging endpoint](https://github.com/aws/amazon-ecs-service-connect-agent/blob/main/README.md?plain=1#L183) on SC Agent to fine tune what all components will become debug level